File: Sem_3/Functional Programming/Tugas_praktikum/bbsort.py
# sorting bubblesort

def pindah_nilai(x,y):
    return y, x
  
def cek_nilai(list_angka,a):
  if a == len(list_angka) - 1:
    return list_angka
  if list_angka[a] > list_angka[a+1]:
    list_angka[a], list_angka[a+1] = pindah_nilai(list_angka[a], list_angka[a+1])
  return cek_nilai(list_angka, a+1)
    
def bubble_rekursive(list_angka,b):
  if b == 0:
    return list_angka
  list_angka = cek_nilai(list_angka, 0)
  return bubble_rekursive(list_angka, b-1)
  
def bubble(list_angka):
  if len(list_angka) <= 1:
    return list_angka
  else:
    return bubble_rekursive(list_angka, len(list_angka))

list_angka = [1,4,7,2,5,8,3,6,9]
hasil_sorting = bubble(list_angka)
print(hasil_sorting)
  

File: Sem_3/Functional Programming/Tugas_praktikum/bbsort.dart
// sorting bubblesort

List<int> pindahNilai(int x, int y){
  return [y,x];
}

List<int> cekNilai(List<int> listAngka, int a){
  if(a == listAngka.length - 1){
    return listAngka;
  }
  if (listAngka[a] > listAngka[a+1]){
    listAngka.setAll(a, pindahNilai(listAngka[a], listAngka[a+1]));
  }
  return cekNilai(listAngka, a+1);
}

List<int> bubbleRekursive(List<int> listAngka, int b){
  if(b == 0){
    return listAngka;
  }
  listAngka = cekNilai(listAngka, 0);
  return bubbleRekursive(listAngka, b-1);
} 

List<int> bubble(listAngka){
  if (listAngka.length <= 1){
    return listAngka;
  }
  else{
    return bubbleRekursive(listAngka, listAngka.length);
  }
}

void main(){
  List<int> listAngka = [1,4,7,2,5,8,3,6,9];
  List<int> hasilSorting = bubble(listAngka);
  print(hasilSorting);
}

